Duplicati/Library/Backend/Rclone/Rclone.cs

```diff
@@ -0,0 +1,242 @@
+            remote_repo = "remote";
+            remote_path = "C:\\temp\\backup";
+            opt_rclone = "";
+            rclone_executable = "rclone";
+            /*should check here if program is installed */
+
+            if (options.ContainsKey(OPTION_LOCAL_REPO))
+                local_repo = options[OPTION_LOCAL_REPO];
+
+            if (options.ContainsKey(OPTION_REMOTE_REPO))
+                remote_repo = options[OPTION_REMOTE_REPO];
+
+            if (options.ContainsKey(OPTION_REMOTE_PATH))
+                remote_path = options[OPTION_REMOTE_PATH];
+
+            if (options.ContainsKey(OPTION_RCLONE))
+                opt_rclone = options[OPTION_RCLONE];
+
+            if (options.ContainsKey(OPTION_RCLONE_EXECUTABLE))
+                rclone_executable = options[OPTION_RCLONE_EXECUTABLE];
+
+            Console.Error.WriteLine(string.Format("Constructor {0}: {1}:{2} {3}", local_repo, remote_repo, remote_path, opt_rclone));
+
+        }
+
+        #region IBackendInterface Members
+
+        public string DisplayName
+        {
+            get { return Strings.Rclone.DisplayName; }
+        }
+
+        public string ProtocolKey
+        {
+            get { return "rclone"; }
+        }
+
+        public bool SupportsStreaming
+        {
+            get { return false; }
+        }
+
+        private string RcloneCommandExecuter(String command, String arguments)
+        {
+            System.Diagnostics.ProcessStartInfo psi = new System.Diagnostics.ProcessStartInfo();
+            psi.Arguments = arguments;
+            psi.CreateNoWindow = true;
+            psi.FileName = command;
+            psi.RedirectStandardError = true;
+            psi.RedirectStandardInput = true;
+            psi.RedirectStandardOutput = true;
+            psi.UseShellExecute = false;
+            psi.WindowStyle = System.Diagnostics.ProcessWindowStyle.Hidden;
+
+#if DEBUG
+            psi.CreateNoWindow = false;
+            psi.WindowStyle = System.Diagnostics.ProcessWindowStyle.Normal;
+#endif
+            System.Diagnostics.Process p = null;
+            try
+            {
+#if DEBUG
+                Console.Error.WriteLine(String.Format("command executing: {0} {1}", psi.FileName, psi.Arguments));
+#endif
+                p = System.Diagnostics.Process.Start(psi);
+            }
+
+            catch (System.ComponentModel.Win32Exception ex)
+            {
+                throw new Exception(String.Format("Program \"{0}\" does not exist", command), ex);
+            }
+
+            string error = p.StandardError.ReadToEnd();
+            p.WaitForExit();
+            if (error.Length > 0) {
+                if (error.Contains("directory not found"))
+                    throw new FolderMissingException(error);
+                if (error.Contains("didn't find section in config file"))
+                    throw new Exception(String.Format("Missing config file? {0}", error));
+            }
+            return p.StandardOutput.ReadToEnd();
```

Comment:
Not sure how much output there is here, but if there is a lot of output, the process could deadlock after filling the stdout buffer, causing it to hang and never trigger `WaitForExit` ?